### DISCUSSION TITLE
Getting ip address of of source in server

### DISCUSSION THREAD

**Discussion by 03psb** [ANLAB-KAIST/KENSv3#197] [Project Q&A 2. Transport Layer (TCP)] [answered]
I'm working on ACCEPT part of the code, and I figured out it is hard to find the ip address of the client. TCP packets doesn't contain informations of ip address. and the following Tip: Figuring out the source IP address to use (https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Figuring-out-the-source-IP-address-to-use) was not helpful, because I found this method doesn't work on finding address of source. But anyway accept function should return the address of client. I wonder how?

**Comment by Pusnow** [comment, marked as answer]
The `packet` of `packetArrived()` contains all addresses including MAC and IP. You can parse the `packet` to retrieve source IP address.

**Reply by 03psb** [reply]
I checked the packet class in E_packet.cpp, I found no sign of method doing that or field that contains potential IP. Can you be more specific please?

**Reply by Pusnow** [reply]
You should parse packet's data according to protocol headers (Ethernet, IPv4, and TCP).
To understand how a packet is structured, refer to the textbook pp. 260-263, 360-363, 508-518.
